mongo/integration/cursor_test.go

```diff
@@ -203,6 +204,55 @@ func TestCursor(t *testing.T) {
 			assert.True(mt, ok, "expected mongo.CommandError, got: %T", err)
 			assert.Equal(mt, failpointData.ErrorCode, mongoErr.Code, "expected code %v, got: %v", failpointData.ErrorCode, mongoErr.Code)
 		})
+
+		cliOpts := options.Client().SetTimeout(2 * time.Second)
+		mtOpts := mtest.NewOptions().ClientOptions(cliOpts)
+		mt.RunOpts("deferred Close uses Timeout", mtOpts, func(mt *mtest.T) {
+			initCollection(mt, mt.Coll)
+
+			// Find with batchSize 2 so All will run getMore for next 3 docs and error.
+			cur, err := mt.Coll.Find(context.Background(), bson.D{},
+				options.Find().SetBatchSize(2))
+			assert.Nil(mt, err, "Find error: %v", err)
+
+			// Create a context and immediately cancel it.
+			canceledCtx, cancel := context.WithCancel(context.Background())
+			cancel()
+
+			// Clear "insert" and "find" events.
+			mt.ClearEvents()
+
+			// Call All with the canceled context and expect context.Canceled.
+			var docs []bson.D
+			err = cur.All(canceledCtx, &docs)
+			assert.NotNil(mt, err, "expected error for All, got nil")
+			assert.True(mt, errors.Is(err, context.Canceled),
+				"expected context.Canceled error, got %v", err)
+
+			// Assert that a "getMore" command was sent and failed (Next used the
+			// canceled context).
+			stEvt := mt.GetStartedEvent()
+			assert.NotNil(mt, stEvt, `expected a "getMore" started event, got no event`)
+			assert.Equal(mt, stEvt.CommandName, "getMore",
+				`expected a "getMore" started event, got %q`, stEvt.CommandName)
+			fEvt := mt.GetFailedEvent()
+			assert.NotNil(mt, fEvt, `expected a successful "getMore" event, got no event`)
+			assert.Equal(mt, fEvt.CommandName, "getMore",
+				`expected a successful "getMore" event, got %q`, fEvt.CommandName)
+
+			time.Sleep(100 * time.Millisecond)
```

Comment:
```suggestion
```

IIUC there is no need to sleep. `cur.All()` will return after `killCursors` completes.